ext/standard/dns.c

```diff
@@ -415,13 +413,16 @@ PHP_FUNCTION(dns_check_record)
 #endif
 
 	RETVAL_TRUE;
-	i = php_dns_search(handle, hostname, C_IN, type, ans, sizeof(ans));
+	i = php_dns_search(handle, hostname, C_IN, type, answer.qb2, sizeof answer);
+	php_dns_free_handle(handle);
 
 	if (i < 0) {
 		RETVAL_FALSE;
```

Comment:
```suggestion
		RETURN_FALSE;
```
Avoid reading answer on failure.